client/post-editor/editor-media-advanced/index.jsx

```diff
@@ -0,0 +1,40 @@
+/**
+ * External depencencies
+ */
+import React, { PropTypes } from 'react';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { toggleEditorMediaAdvanced } from 'state/ui/editor/media/actions';
+import Dialog from 'components/dialog';
+
+function EditorMediaAdvanced( { visible, toggleVisible } ) {
+	return (
+		<Dialog isVisible={ visible } onClose={ toggleVisible }>
+			WIP
+		</Dialog>
+	);
+}
+
+EditorMediaAdvanced.propTypes = {
+	visible: PropTypes.bool,
+	toggleVisible: PropTypes.func
+};
+
+EditorMediaAdvanced.defaultProps = {
+	visible: false,
+	toggleVisible: () => {}
+};
+
+export default connect( ( state ) => {
+	return {
+		visible: state.ui.editor.media.advanced
+	};
+}, ( dispatch ) => {
+	return bindActionCreators( {
+		toggleVisible: toggleEditorMediaAdvanced
+	}, dispatch );
+} )( EditorMediaAdvanced );
```

Comment:
>this is another case where to me the `return`s and the brackets make it harder for me to see what is happening.

The concern I'd have with your proposal is that it would quickly become difficult to read once more than one or two props are being returned. It'd be nice if we can standardize on a consistent approach. Another I've seen is keeping the callbacks at the same indentation level, optionally naming the functions for more clarity:

```jsx
export default connect(
	function mapStateToProps( state ) {
		return {
			visible: state.ui.editor.media.advanced
		};
	},
	function mapDispatchToProps( dispatch ) {
		return bindActionCreators( {
			toggleVisible: toggleEditorMediaAdvanced
		}, dispatch );
	}
)( EditorMediaAdvanced );
```